packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -315,48 +274,58 @@ export function useGridEditRows(apiRef: GridApiRef) {
     [apiRef],
   );
 
-  const handleDoubleClick = React.useCallback(
+  const handleCellEditExit = React.useCallback(
+    (params: GridCellParams, event?: React.SyntheticEvent) => {
+      // TODO check if its propagation was stopped
+      setCellMode(params.id, params.field, 'view');
+
+      // When dispatched by the document, the event is not passed
+      if (!event || !isKeyboardEvent(event)) {
+        return;
+      }
+
+      if (isCellEditCommitKeys(event.key)) {
+        apiRef.current.publishEvent(GRID_CELL_NAVIGATION_KEY_DOWN, params, event);
+        return;
+      }
+      if (isEscapeKey(event.key) || isDeleteKeys(event.key)) {
```

Comment:
Time to stop using abstractions that don't solve problems? (we need isDeleteKeys)

```suggestion
      if (event.key === 'Enter' || isDeleteKeys(event.key)) {
```

Reply:
Am it should be `Escape` instead of `Enter`.

Reply:
This `isDeleteKeys`  and other similar abstractions we could bring to inside the hook. The probability of using it again in another feature is very small.